presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -376,8 +377,13 @@ public static void renameDirectory(HdfsEnvironment hdfsEnvironment, String schem
     public static Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        // TODO: this should use Hadoop UserGroupInformation
-        String temporaryPrefix = "/tmp/presto-" + StandardSystemProperty.USER_NAME.value();
+        String temporaryPrefix;
+        try {
+            temporaryPrefix = "/tmp/presto-" + UserGroupInformation.getCurrentUser().getUserName();
```

Comment:
@billonahill. I wanted to change the code  it as you suggest. 
But as I read the codebase it seems that temporary file is not created in `/tmp` 
but in `$targetPath/tmp/`. This is by definition on the same filesystem as `$targetPath`. I will debug yet - to verify - but it seems like you should not have the problem you are describing. Am I missing somethign?